src/Evolution/Executables/SingleBH/EvolveSingleBH.hpp

```diff
@@ -0,0 +1,183 @@
+#include "AlgorithmSingleton.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Printf.hpp"
+/// [executable_example_includes]
+
+/// [executable_example_options]
+namespace OptionTags {
+struct Name {
+  using type = std::string;
+  static constexpr OptionString help{"A name"};
+};
+}  // namespace OptionTags
+/// [executable_example_options]
+
+/// [executable_example_action]
+namespace Actions {
+struct PrintMessage {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) {
+    Parallel::printf("Hello %s from process %d on node %d!\n",
+                     Parallel::get<OptionTags::Name>(cache),
+                     Parallel::my_proc(), Parallel::my_node());
+  }
+};
+}  // namespace Actions
+/// [executable_example_action]
+
+/// [executable_example_singleton]
+template <class Metavariables>
+struct HelloWorld {
+  using const_global_cache_tag_list = tmpl::list<OptionTags::Name>;
+  using chare_type = Parallel::Algorithms::Singleton;
+  using metavariables = Metavariables;
+  using action_list = tmpl::list<>;
+  using initial_databox = db::DataBox<tmpl::list<>>;
+  using options = tmpl::list<>;
+  static void initialize(Parallel::CProxy_ConstGlobalCache<
+                         Metavariables>& /* global_cache */) noexcept {}
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache) noexcept;
+};
+
+template <class Metavariables>
+void HelloWorld<Metavariables>::execute_next_phase(
+    const typename Metavariables::Phase /* next_phase */,
+    Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache) noexcept {
+  Parallel::simple_action<Actions::PrintMessage>(
+      Parallel::get_parallel_component<HelloWorld>(
+          *(global_cache.ckLocalBranch())));
+}
+/// [executable_example_singleton]
+
+/// [executable_example_metavariables]
+struct EvolutionMetavars {
+  static constexpr int Dim = 3;
+  using system = GeneralizedHarmonic::System<Dim>;
+
+  using analytic_solution = gr::Solutions::KerrSchild; // FIXME
+  using analytic_solution_tag = OptionTags::AnalyticSolution<analytic_solution>;
+  using analytic_variables_tags = typename system::variables_tags;
+  //using temporal_id = Tags::TimeID;
+  static constexpr bool local_time_stepping = false;
+
+/*
+  // FIXME - where are the fluxes?
```

Comment:
Part of PR #1358 implements the correct upwind flux, but I haven't pushed that as a separate PR because I liked the suggestion to first demonstrate it with a KerrSchild evolution, to make sure it's doing the right thing